cmd/fetch_commit.go

```diff
@@ -109,7 +113,18 @@ func validateGitSource(ctx context.Context, imageRef, publicKey string) error {
 		return err
 	}
 
+	invalidPredicateMaterialCount := 0
 	for _, att := range validatedImage.Attestations {
+		if att.Predicate.Materials == nil {
```

Comment:
Although this achieves the job, I'm not sure if teaching `fetch_commit.go` about `.Predicate.Materials` is the best approach. What if we decide to look for git info differently in `NewGitSource` ? It won't be obvious that we also have to update here.

Can we just check the error from `att.NewGitSource()` a few lines below?


```go
gitSourceFound := false

for _, att := range validatedImage.Attestations {
  ...
  gitSource, err := att.NewGitSource()
  if err != nil {
    log.Debugf(...)
    continue
  }
  gitSourceFound = true
  ...
}

if !gitSourceFound {
  return errors.New(...)
}
return nil
```